c/helper.c

```diff
@@ -13,10 +13,10 @@ int buffer_append(void *ext, unsigned char *str, size_t size)
 {
 	buffer_t *b = ext;
 	int new_size, new_used;
-	char *tmp;
+	unsigned char *tmp;
 
 	new_used = b->used + size;
-	for (new_size = b->size || 8; new_size < new_used; new_size *= 2);
+	for (new_size = 1; new_size < new_used; new_size *= 2);
```

Comment:
This is a non behaviour changing fix.   We don't know what b->size is, but 8 is always "true" and "true" is 1.  Try running this:

```
#include <stdio.h>

main() {
   int x = 5;
   int y = x || 8;
   printf("%d\n", y);
}
```